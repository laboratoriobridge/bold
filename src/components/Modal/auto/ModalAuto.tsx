import React, { memo, useEffect, useState } from 'react'

import { Button, ButtonProps } from '../../Button'
import { HFlow } from '../../HFlow'
import { Modal, ModalDepthLevel, ModalProps } from '../Modal'
import { ModalBody } from '../ModalBody'
import { ModalFooter } from '../ModalFooter'

export type ButtonAction = ButtonProps & { label?: React.ReactNode }

export interface ModalAutoProps {
  actions?: ButtonAction[]
  size?: ModalProps['size']

  /**
   * @description allows you to customize the depth of the container and the backdrop of the modal
   * @default 1 - the lowest possible value
   */
  depthLevel?: ModalDepthLevel

  /**
   * @description allows you to remove the document's overflow property when a modal is closed
   * @default true
   */
  manageOverflow?: boolean

  /**
   * Specify whether the `onClose` prop should called when backdrop is clicked.
   * @default true
   */
  closeOnBackdropClick?: boolean

  render(renderProps: ModalAutoRenderProps): React.ReactNode
  dispose(): void
  onClose?(): any
}

export interface ModalAutoState {
  open: boolean
}

export interface ModalAutoRenderProps {
  close(): void
}

export const ModalAuto = memo((props: ModalAutoProps) => {
  const { actions, size, render, dispose, onClose, closeOnBackdropClick, ...rest } = props

  const [isOpen, setIsOpen] = useState(false)

  useEffect(() => {
    setIsOpen(true)
  }, [])

  const close = () => {
    onClose && onClose()
    setIsOpen(false)

    // Dispose with timeout to preserve closing transition
    setTimeout(dispose, 500)
  }

  const handleAction = (action: ButtonProps) => (e) => {
    action.onClick && action.onClick(e)
    close()
  }

  return (
    <Modal open={isOpen} size={size} onClose={close} closeOnBackdropClick={closeOnBackdropClick} {...rest}>
      <ModalBody>{render({ close })}</ModalBody>
      {actions && (
        <ModalFooter>
          <HFlow justifyContent='flex-end'>
            {actions.map((action, idx) => {
              const { label, ...rest } = action
              return (
                <Button key={idx} style={{ minWidth: 144 }} {...rest} onClick={handleAction(action)}>
                  {label}
                </Button>
              )
            })}
          </HFlow>
        </ModalFooter>
      )}
    </Modal>
  )
})
