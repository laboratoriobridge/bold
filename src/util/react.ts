import { MutableRefObject, Ref } from 'react'

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}
