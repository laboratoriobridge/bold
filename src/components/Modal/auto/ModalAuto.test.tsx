import { fireEvent, render } from '@testing-library/react'
import React from 'react'

import { ModalAuto } from './ModalAuto'

jest.mock('../../../util/string')

jest.useFakeTimers()

it('should render correctly', () => {
  const component = (
    <ModalAuto
      title='Modal auto'
      dispose={jest.fn()}
      render={() => <span>Body</span>}
      size='small'
      actions={{
        primarySlot: {
          label: 'Confirm',
          onClick: jest.fn(),
        },
      }}
    />
  )
  const { rerender } = render(component)
  rerender(component)
  expect(document.body).toMatchSnapshot()
})

it('should render correctly with depth', () => {
  const component = (
    <ModalAuto
      title='Modal auto'
      dispose={jest.fn()}
      render={() => <span>Body</span>}
      depthLevel={3}
      actions={{
        primarySlot: {
          label: 'Confirm',
          onClick: jest.fn(),
        },
      }}
    />
  )
  const { rerender } = render(component)
  rerender(component)
  expect(document.body).toMatchSnapshot()
})

it(`should render correctly without 'overflow hidden' prop in document classList`, () => {
  const component = (
    <ModalAuto
      title='Modal auto'
      dispose={jest.fn()}
      render={() => <span>Body</span>}
      manageOverflow={false}
      actions={{
        primarySlot: {
          label: 'Confirm',
          onClick: jest.fn(),
        },
      }}
    />
  )
  const { rerender } = render(component)
  rerender(component)
  expect(document.body.classList).not.toContainEqual('oveflow')
  expect(document.body).toMatchSnapshot()
})

it('should open when mounted', () => {
  const component = <ModalAuto title='Modal auto' dispose={jest.fn()} render={() => <span>Body</span>} />
  const { rerender } = render(component)
  rerender(component)
  expect(document.body.querySelector('[role="dialog"]')).toBeTruthy()
})

it('should close modal when a button is clicked', () => {
  const confirmHandler = jest.fn()
  const component = (
    <ModalAuto
      title='Modal auto'
      dispose={jest.fn()}
      render={() => <span>Body</span>}
      actions={{
        primarySlot: {
          label: 'Confirm',
          kind: 'primary',
          onClick: confirmHandler,
        },
        secondarySlot: { label: 'Cancel' },
      }}
    />
  )

  const { getByText, rerender } = render(component)
  rerender(component)

  fireEvent.click(getByText('Confirm'))
  expect(confirmHandler).toHaveBeenCalled()
  expect(document.body.querySelector('[role="dialog"]')).toBeFalsy()
})

it('should pass render props', () => {
  const renderModal = jest.fn()
  render(<ModalAuto title='Modal auto' dispose={jest.fn()} render={renderModal} />)
  expect(renderModal.mock.calls[0][0].close).toEqual(expect.any(Function))
})
