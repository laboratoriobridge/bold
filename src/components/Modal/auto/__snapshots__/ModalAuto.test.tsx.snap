// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-7 {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  pointer-events: none;
  z-index: 1100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.emotion-10 {
  overflow: hidden;
}

.emotion-6 {
  border: 1px solid #D3D4DD;
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14);
  border-radius: 4px;
  background-color: #FFFFFF;
  min-width: 520px;
  pointer-events: auto;
  max-height: 80vh;
  overflow: auto;
  width: 520px;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  float: right;
  margin-top: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-1 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-2 {
  padding: 2.5rem;
}

.emotion-5 {
  background-color: #F0F0F5;
  padding: 1rem;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-4 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 1rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-3 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: #FFFFFF;
  border: 1px solid #696979;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 1rem;
  min-width: 144px;
}

.emotion-3 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-3:not(:disabled):hover {
  background-color: #F0F0F5;
}

.emotion-8 {
  position: fixed;
  z-index: 1050;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(211,212,221,0.7);
}

.emotion-9 {
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  opacity: 0;
  opacity: 0;
}

<body
  class="emotion-10"
>
  <div />
  <div
    class="emotion-9"
  >
    <div
      class="emotion-7"
    >
      <div
        aria-modal="true"
        class="emotion-6"
        role="dialog"
      >
        <button
          aria-label="Close"
          class="emotion-1"
          title="Close"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                clip-rule="evenodd"
                d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0v.001z"
              />
            </svg>
          </span>
        </button>
        <div
          class="emotion-2"
        >
          <span>
            Body
          </span>
        </div>
        <div
          class="emotion-5"
        >
          <div
            class="emotion-4"
          >
            <button
              class="emotion-3"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-8"
      data-testid="backdrop"
    />
  </div>
</body>
`;

exports[`should render correctly with depth 1`] = `
.emotion-10 {
  overflow: hidden;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  float: right;
  margin-top: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-1 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-2 {
  padding: 2.5rem;
}

.emotion-5 {
  background-color: #F0F0F5;
  padding: 1rem;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-4 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 1rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-3 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: #FFFFFF;
  border: 1px solid #696979;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 1rem;
  min-width: 144px;
}

.emotion-3 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-3:not(:disabled):hover {
  background-color: #F0F0F5;
}

.emotion-9 {
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  opacity: 0;
  opacity: 0;
}

.emotion-7 {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  pointer-events: none;
  z-index: 1500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.emotion-6 {
  border: 1px solid #D3D4DD;
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14);
  border-radius: 4px;
  background-color: #FFFFFF;
  min-width: 520px;
  pointer-events: auto;
  max-height: 80vh;
  overflow: auto;
  width: 900px;
}

.emotion-8 {
  position: fixed;
  z-index: 1450;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(211,212,221,0.7);
}

<body
  class="emotion-10"
>
  <div />
  <div
    class="emotion-9"
  >
    <div
      class="emotion-7"
    >
      <div
        aria-modal="true"
        class="emotion-6"
        role="dialog"
      >
        <button
          aria-label="Close"
          class="emotion-1"
          title="Close"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                clip-rule="evenodd"
                d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0v.001z"
              />
            </svg>
          </span>
        </button>
        <div
          class="emotion-2"
        >
          <span>
            Body
          </span>
        </div>
        <div
          class="emotion-5"
        >
          <div
            class="emotion-4"
          >
            <button
              class="emotion-3"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-8"
      data-testid="backdrop"
    />
  </div>
</body>
`;

exports[`should render correctly without 'overflow hidden' prop in document classList 1`] = `
.emotion-7 {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  pointer-events: none;
  z-index: 1100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  float: right;
  margin-top: 0.5rem;
  margin-right: 0.5rem;
}

.emotion-1 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-2 {
  padding: 2.5rem;
}

.emotion-5 {
  background-color: #F0F0F5;
  padding: 1rem;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-4 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 1rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-3 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: #FFFFFF;
  border: 1px solid #696979;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.75rem - 1px) 1rem;
  min-width: 144px;
}

.emotion-3 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-3:not(:disabled):hover {
  background-color: #F0F0F5;
}

.emotion-8 {
  position: fixed;
  z-index: 1050;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(211,212,221,0.7);
}

.emotion-9 {
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  opacity: 0;
  opacity: 0;
}

.emotion-6 {
  border: 1px solid #D3D4DD;
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14);
  border-radius: 4px;
  background-color: #FFFFFF;
  min-width: 520px;
  pointer-events: auto;
  max-height: 80vh;
  overflow: auto;
  width: 900px;
}

<body
  class=""
>
  <div />
  <div
    class="emotion-9"
  >
    <div
      class="emotion-7"
    >
      <div
        aria-modal="true"
        class="emotion-6"
        role="dialog"
      >
        <button
          aria-label="Close"
          class="emotion-1"
          title="Close"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="emotion-0"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                clip-rule="evenodd"
                d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0v.001z"
              />
            </svg>
          </span>
        </button>
        <div
          class="emotion-2"
        >
          <span>
            Body
          </span>
        </div>
        <div
          class="emotion-5"
        >
          <div
            class="emotion-4"
          >
            <button
              class="emotion-3"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-8"
      data-testid="backdrop"
    />
  </div>
</body>
`;
